src/SingleCellAnnotator.imjoy.html

```diff
@@ -847,9 +840,58 @@
     },
     startRedrawAnnotation() {
       this.current_image_data.mask_adjusted = true;
-      this.vector_source.clear()
+      if (this.single_cell_mode){
+        this.vector_source.clear()
+      };
       this.startMaskAnnotation(this.vector_source, this.freehand_mode)
     },
+
+    async restoreAnnotation() {
+      if (!this.selected_single_mask){
+        const features = (new GeoJSON()).readFeatures(this.current_image_data.mask_geojson)
+      }else{
+        const features = (new GeoJSON()).readFeatures(this.selected_single_mask)
```

Comment:
Why do we need to serialize the features in order to keep in `this.selected_single_mask`, I think you can just assign the value `this.select.getFeatures()` to `this.selected_single_mask` below and assign it directly here.